---
title: "**Exercise 1**"
author: "Carlos López Elorduy"
date: "5/15/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction
**Title:** Comparison between Saccharomyces cerevisiae HS 10, HS 34 and HS 12, HS 37

**Organism:** Saccharomyces cerevisiae.

**Experiment type:** Expression profiling by array.

**Status:** Public on Dec 12, 2008.

## Summary
Gene expression of different flocculent (HS 10 and HS 34) and powdery (HS 12 and HS 37) yeast strains compared to each other during exponential and stationary growth phase was analysed.

The isolation of RNA was done by disruption of the cells under liquid nitrogen using mortar and pistil and then the Qiagen RNeasy Midi Kit with some modifications within the manufacturers´ protocol and the Qiagen RNase-Free DNase Set were applied.

6 µg of the total RNA per sample was used for each microarray experiments. The indirect labelling by the tyramide-signal-amplification method (MicromaxTM TSATM labelling and detection Kit from Perkin Elmer life sciences) was used to increase the Cy3 and Cy5 signals of microarray detection.

Each cDNA containing Biotin- and Fluorescein-nucleotides respectively was purified with a QIAquick PCR purification kit and suspended in 11 µl of the formamide containing hybridization buffer. The slides were hybridized at 42°C over night under a cover slip.

The microarrays were scanned by the Axon 4000B scanner; image intensity data were extracted and analysed with GenePix® Pro 6.0 software.

Data from different scans of Dye-swap experiment were extracted by GenePix Pro 6.0 software, normalized and united. An outliertest has been applied in order to find outliers amongst the gene replicats. Subsequently, a t-test (1% and 5% probability of error) has been used in order to find regulated genes.
Keywords: sorted yeast cells

## Experimental design
Microarray experiments were realised with dye swap.

## Citations
  Heine F, Stahl F, Sträuber H, Wiacek C et al. Prediction of flocculation ability of brewing yeast inoculates by flow cytometry, proteome analysis, and mRNA profiling. Cytometry A 2009 Feb;75(2):140-7. PMID: 19072835

## Data description
We have two files:

  - The **targets** file
  - The **expression matrix** file

These files can be either imported manually using the Rstudio tools or using the following commands:
```{r}
targets<-read.csv("targets_12369.txt",header = T, sep = "\t")
myExpress<-as.matrix(read.table("matrix_12369.txt", header=T, sep="\t", row.names=1, as.is=T))
```

### Targets
```{r}
show(targets)
```

### Expression matrix
```{r}
dim(myExpress)
summary(myExpress)
```

## ExpressionSet creation

The expression set will contain all the relevant information about our study, starting by the expression matrix:
```{r}
library("Biobase")
myEset<-ExpressionSet(myExpress)
```

The next information we have to add is the information stored in the targets.
```{r}
columnDesc<-data.frame(labelDescription=c("SampleName", "Experiment", "Color"))
myAnnotDF <- new("AnnotatedDataFrame", data=targets, varMetadata= columnDesc)
show(myAnnotDF)
```
We create an array of the length of the total number of genes (576 in my case)
```{r}
myGenes <-  paste0("gene",1:576)
```

We store the annotated information dataframe the following way, so that we can add it to the expression set.
```{r}
phenoData(myEset)<-myAnnotDF
```

```{r}
rownames(pData(myAnnotDF))<-pData(myAnnotDF)$SampleName
myEset <- ExpressionSet(assayData=myExpress, phenoData=myAnnotDF)
```

We finally add the array of genes to the expression set:
```{r}
myEset <- ExpressionSet(assayData=myExpress,phenoData=myAnnotDF,featureNames =myGenes)
show(myEset)
```

I can add also my contact info to the expression set:
```{r}
myInfo=list(myName="Carlos Lopez Elorduy", myLab="Bioinformatics Lab",myContact="carlos.lopez@alum.esci.upf.edu", myTitle="Practical Exercise on ExpressionSets")
```
